templates/base/client/RtagClient.java.hbs

```diff
@@ -0,0 +1,159 @@
+                .post(RequestBody.create(JSON, "{}")).build();
+        String response = client.newCall(request).execute().body().string();
+        return jsonMapper.readValue(response, LoginResponse.class).getToken();
+    }
+
+    {{/if}}
+    {{/each}}
+
+    public static String createState(String origin, String token, ICreateGameRequest request) throws IOException {
+        String response = client.newCall(new Request.Builder().url(String.format("%s/new", origin))
+                .post(RequestBody.create(JSON, "{}")).header("Authorization", "Bearer " + token).build()).execute()
+                .body().string();
+        return jsonMapper.readValue(response, CreateStateResponse.class).getStateId();
+    }
+
+    public static RtagClient connect(String host, String token, String stateId, StateChangeRunnable onStateChange)
+            throws URISyntaxException {
+        RtagClient client = new RtagClient();
+        WebSocketClient socket = new WebSocketClient(new URI(String.format("ws://%s/%s", host, stateId))) {
+
+            @Override
+            public void onOpen(ServerHandshake handshakedata) {
+                this.send(token);
+            }
+
+            @Override
+            public void onMessage(String message) {
+                onMessage(ByteBuffer.wrap(message.getBytes()));
+            }
+
+            @Override
+            public void onMessage(ByteBuffer message) {
+                try {
+                    byte[] data = message.array();
+                    Message msg = msgPackMapper.readValue(data, Message.class);
+                    if (msg.getType().equals("response")) {
+                        ResponseMessage responseMsg = msgPackMapper.readValue(data, ResponseMessage.class);
+                        client.callbacks.get(responseMsg.getMsgId()).run(responseMsg.getError());
+                        client.callbacks.remove(responseMsg.getMsgId());
+                    } else if (msg.getType().equals("state")) {
+                        StateMessage stateMsg = msgPackMapper.readValue(data, StateMessage.class);
+                        onStateChange.run(stateMsg.getState());
+                    } else {
+                        throw new Exception(String.format("Unknown message type: %s", msg.getType()));
+                    }
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+
+            @Override
+            public void onClose(int code, String reason, boolean remote) {
+                System.out.println("connection closed");
+            }
+
+            @Override
+            public void onError(Exception ex) {
+                ex.printStackTrace();
+            }
+
+        };
+        socket.connect();
+        client.setSocket(socket);
+        return client;
+    }
+
+    {{#each methods}}
+    public void {{@key}}({{makeRequestName @key}} request, CallbackRunnable cb) {
```

Comment:
It looks like the latest version of this method returns a Result. Need to convert this to make the logic match.

Reply:
Futures